csharp-mode.el

```diff
@@ -285,12 +284,27 @@ casts and declarations are fontified.  Used on level 2 and higher."
           ;; merge with cc-mode defaults
           (c-lang-const c-basic-matchers-after)
 
+          ;; Hack: Last prop-identifier
+          `(("\\.\\([A-Z][A-Za-z0-9_]+\\)" 1 font-lock-variable-name-face t))
+          `(("^using\\s *\\([A-Z][A-Za-z0-9_]+\\);" 1 font-lock-variable-name-face t))
+          `(("^namespace\\s *\\([A-Z][A-Za-z0-9_]+\\)\\s *" 1 font-lock-variable-name-face t))
+
           ;; function names
-          `(("\\.\\([A-Za-z0-9_]+\\)[<(]" 1 font-lock-function-name-face t))
+          `(("\\([A-Za-z0-9_]+\\)\\(<[a-zA-Z0-9, ]+>\\)?(" 1 font-lock-function-name-face t))
+
+          ;; class names with inheritance
+          `(("\\<\\([A-Z][A-Za-z0-9_]+\\)\\s *:" 1 font-lock-type-face t))
+
+          ;; Single identifier in attribute
+          `(("\\[\\([A-Z][A-Za-z0-9_]*\\)\\][^;]" 1 font-lock-variable-name-face t))
+
+          ;;  Types after 'new'
+          `((,(concat "new\\s *\\([" (c-lang-const c-symbol-chars) "]+\\)")
```

Comment:
Any reason why you are using `\\([A-Z][A-Za-z0-9_]+\\)` for types elsewhere, but here you are using `\\([" (c-lang-const c-symbol-chars) "]+\\)` ?

Also: Should we define our own constant, for instance `csharp-re-typename` ?

Reply:
Speaking of which... I see you are using a variation of these 2 regexps:

* `\\([A-Z][A-Za-z0-9_]+\\)`
* `\\([A-Z][A-Za-z0-9_]*\\)`

The first variant forces a type-name to have to be 2-characters or more, while the second allows single-character type-names, like `T`. I think this is the more correct regexp to use, unless you have specific reasons not to.

And seeing inconsistencies like this in the code, again makes me think defining this as a single constant is a good idea.

Reply:
Fixed in 91c0f3c - do you agree?

Reply:
Yeah, I saw that when going over it again. Thanks for pointing it out. Should be clearer now. These inconsistencies come from hacking on small variations over and over again 😄 

I think not using the `c-symbol-chars` is the correct way to go, since we have control over that, and it is a simple regex.

Reply:
Looks good to me now :+1: